ntbs-service/Services/AuthorizationService.cs

```diff
@@ -37,78 +36,99 @@ public async Task<bool> IsUserAuthorizedToManageAlert(ClaimsPrincipal user, Aler
+            }
+            return true;
+            
         }
 
         public async Task<PermissionLevel> GetPermissionLevelForNotificationAsync(ClaimsPrincipal user,
-            Notification notification, NotificationGroup group = null)
+            Notification notification)
         {
-            if (await CanEditNotificationAsync(user, notification))
+            if (_filter == null)
+            {
+                _filter = await GetUserPermissionsFilterAsync(user);
+            }
+            
+            if (UserHasDirectRelationToNotification(notification))
             {
                 return PermissionLevel.Edit;
             }
 
-            if (await CanEditLinkedNotificationAsync(user, group))
+            if (UserHasDirectRelationToLinkedNotification(notification?.Group?.Notifications))
             {
                 return PermissionLevel.ReadOnly;
             }
 
             return PermissionLevel.None;
         }
-
-        public async Task<bool> CanEditLinkedNotificationAsync(ClaimsPrincipal user, NotificationGroup group)
+        
+        private bool UserHasDirectRelationToLinkedNotification(IEnumerable<Notification> linkedNotifications)
         {
-            return group?.Notifications.Select(n => CanEditNotificationAsync(user, n).Result).Any(x => x == true) ??
-                   false;
+            return linkedNotifications.Select(UserHasDirectRelationToNotification).Any(x => x == true);
         }
 
-        public async Task<bool> CanEditNotificationAsync(ClaimsPrincipal user, Notification notification)
+        private bool UserHasDirectRelationToNotification(Notification notification)
         {
-            var tbServiceCode = notification.Episode.TBServiceCode;
-            var tbServicePhecCode = notification.Episode.TBService?.PHECCode;
-            var locationPhecCode = notification.PatientDetails.PostcodeLookup?.LocalAuthority?.LocalAuthorityToPHEC?.PHECCode;
-            return await AuthorizeUserAccessForTbServiceAndPhecCodes(user, tbServiceCode, locationPhecCode, tbServicePhecCode);
+            return UserBelongsToTbServiceOfNotification(notification) || UserBelongsToTreatmentPhecOfNotification(notification) ||
+                UserBelongsToResidencePhecOfNotification(notification);
         }
 
-        private async Task<bool> AuthorizeUserAccessForTbServiceAndPhecCodes(ClaimsPrincipal user, string tbServiceCode,
-            string locationPhecCode, string tbServicePhecCode)
+        private bool UserBelongsToTbServiceOfNotification(Notification notification)
         {
-            if (_filter == null)
+            if (_filter.FilterByTBService)
             {
-                _filter = await GetUserPermissionsFilterAsync(user);
+                return _filter.IncludedTBServiceCodes.Contains(notification.Episode.TBServiceCode);
             }
-
-            if (_filter.FilterByTBService)
+            return false;

```

Comment:
Would this be cleaner as an `&&` condition, so:
```
return _filter.FilterByTbService && _filter.IncludedTBServiceCodes.Contains(notification.Episode.TBServiceCode)
```
or, if we drop usage of `FilterByX` like I suggested above:
```
return _filter.UserType == NhsUser && _filter.IncludedTBServiceCodes.Contains(notification.Episode.TBServiceCode)
```

The more I think about it, the more I dislike the name "filter" too... Should we rename it to something like `UserRelations`? Appreciate that's tangential to the change, but perhaps you have a feeling for a good name here, after massaging this code a fair bit :)